### ISSUE TITLE
Suppress time signature when mensuration is present in staffDef/scoreDef element

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#853]
When `@mensur.sign` is present in `staffDef` and `scoreDef`, a time signature should not be displayed as well as the mensuration sign.   In other words `@meter.count`  and `@meter.unit` should be ignored if `@mensur.sign` is present in the element.

Example current rendering of following data:

<img width="941" alt="screen shot 2018-05-07 at 12 26 39 am" src="https://user-images.githubusercontent.com/3487289/39689988-6bd63fc4-518d-11e8-81ce-2d11336dde41.png">

Desired rendering:

<img width="892" alt="screen shot 2018-05-07 at 12 26 49 am" src="https://user-images.githubusercontent.com/3487289/39689977-65558e52-518d-11e8-9f45-a88ce2b5cba9.png">

Notice how `@meter.sym` suppresses the time signature.  The presence of `@mensur.sign` should behave the same way.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-05-07T00:24:07" version="2.0.0-dev-973d948-dirty">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000001561513073">
                <score xml:id="score-0000002100051571">
                    <scoreDef xml:id="scoredef-0000001716342111" midi.bpm="400">
                        <staffGrp xml:id="staffgrp-0000000540802885">
                            <staffDef xml:id="staffdef-0000001365630893" meter.count="2" meter.unit="1" clef.shape="G" clef.line="2" mensur.sign="C" mensur.slash="1" n="1" lines="5">
                                <label xml:id="label-0000000404421984" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001673015352">
                        <measure xml:id="measure-L5" n="1">
                            <staff xml:id="staff-L5F1N1" n="1">
                                <layer xml:id="layer-L5F1N1" n="1">
                                    <note xml:id="note-L6F1" dur="1" oct="4" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <scoreDef xml:id="scoredef-0000000244362211" meter.count="2" meter.unit="2" meter.sym="cut" />
                        <measure xml:id="measure-L7" n="2">
                            <staff xml:id="staff-L7F1N1" n="1">
                                <layer xml:id="layer-L7F1N1" n="1">
                                    <note xml:id="note-L10F1" dur="1" oct="4" pname="d" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <scoreDef xml:id="scoredef-0000000789043163" meter.count="3" meter.unit="1" mensur.sign="O" />
                        <measure xml:id="measure-L11" right="end" n="3">
                            <staff xml:id="staff-L11F1N1" n="1">
                                <layer xml:id="layer-L11F1N1" n="1">
                                    <note xml:id="note-L14F1" dots="1" dur="1" oct="4" pname="e" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
I do not think that having both `@mensur.sign` and `@meter.count/unit` makes sense. What would make sense would be to add `@mensur.num/numbase` on staffDef, so then we would be inline with attributes available on `mensur`.

**Comment by craigsapp** [comment]
That would be OK if `@mensur.num` and `@mensur.numbase` have the same meaning as `@meter.count` and `@meter.unit`.  These do not seem to have the same meaning:

http://music-encoding.org/guidelines/v3/elements/mensur.html

<img width="733" alt="screen shot 2018-05-07 at 12 55 39 am" src="https://user-images.githubusercontent.com/3487289/39691119-55e49b58-5191-11e8-9a83-8ebf08bf9dd4.png">

The descriptions of these to attributes are a bit generic, i.e., a duration ratio of what?


This is for CMN notation where there is an underlying time signature, such as 2/1, that approximates the original mensuration, but it is supposed to be replaced with the original mensuration sign of the music (Cut-C mensuration for the first measure above, for example).  The original mensuration signs encode both the time signature and the tempo, so they have more information that a modern time signature which does not encode the tempo as much (cut-time versus common-time are relics of C and Cut-C mensuration, for example, that also have tempo implications).

**Comment by lpugin** [comment]
You are right. Actually, what should be used is `@proport.num/numbase`, which are already available on staffDef. I am not sure about the status of Verovio with this.

**Comment by rettinghaus** [comment]
Perhaps it does make some sense to have mensur + meter in staffDef. But without notationtype specified, it's hard to know what to make out of it. With `notationtype="mensural"` it should go for mensur, otherwise take only the meter.

**Comment by craigsapp** [comment]
> With `notationtype="mensural"` it should go for mensur, otherwise take only the meter.

That makes sense, but it would be a mostly useless rule that would cause technical problems.  If someone goes to the trouble of putting `mensur.*` attributes in the `staffDef`/`scoreDef`, then they probably want to have the mensuration shown. 

The `mensur*` attributes should instead behave like `meter.sym` since they are closely related concepts.  In other words, if `meter.sym` is present it will  have display priority over the numeric time signature.  Likewise if there is a mensuration, then it would be displayed instead of the time signature.

One particular reason for this is that `@meter.unit` is needed in order to specify the units that `@tstamp` are using, so removing the meter unit would mess up all of the tstamps.

If someone wanted to display the time signature, then they could use XSLT to remove the `mensur.*` attributes before printing with verovio.

This rule would also dictate that `mensur.*` has priority over `meter.sym`, even though this is less obvious.

I can also imagine an alternate rendering style that users might like: display the time signature (or meter.sym) on the staff, and display the mensuration sign above the time signature. This could be done with an option input from verovio, or some sort of attribute could be added to staffDef/scoreDef to indicate this style.  If it were added as an attribute, there could be three states: (1) display only time signature, (2) display only mensuration, (3) display time signature with mensuration sign above it.

**Comment by craigsapp** [comment]
I am currently displaying the mensuration sign, and also giving `@meter.unit` to allow placement of `@tstamp` values.  However, when `@meter.count` is removed to prevent display of the time signature after the mensuration, the MIDI output from verovio is incorrect (it seems that `@meter.count="4"` is used as the default.)

So `@meter.unit` is needed for `@tstamp`s, and `@meter.count` is needed for MIDI generation.

This is related to issue https://github.com/humdrum-tools/verovio-humdrum-viewer/issues/99

This also seems to be true when the notation type is set to mensural (at least when `<measures>` are being used.

**Comment by rettinghaus** [comment]
Proposal: Use `@notationtype` as a switch to turn off display of meter signature, and put the mensural sign in the `<layer>` at the beginning.

**Comment by craigsapp** [comment]
> Proposal: Use @notationtype as a switch to turn off display of meter signature, and put the mensural sign in the <layer> at the beginning.

A problem with this is that in JRP scores, the music is shown in CMN, but mensuration signs are shown instead of time signatures.  So I would not want to show mensural-style notes (diamonds instead if circles for semi-breves), and also only modern-style clefs.

**Comment by lpugin** [comment]
This is solved with 112b8999699ba30045337abf84f4991e18d2337b
